### PR TITLE
Allow large problems to work with stiff ODE solvers in `EnsembleGPUKernel`

### PR DESCRIPTION
**Pull request by utkarsh530**
The PR allows us to solve relatively larger state parameter-parallel models. It is made possible by removing the constraint in StaticArrays.jl, which forces to allocate when size > 14. The problem is mainly in the Linear Algebra routines such as `lu` and `\` .

### CODE REVIEW

File: src/linalg/linsolve.jl [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+for Sa in [(2, 2), (3, 3)]  # not needed for Sa = (1, 1);
+    @eval begin
+        @inline function _linear_solve(::Size{$Sa},
+            ::Size{Sb},
+            a::StaticMatrix{<:Any, <:Any, Ta},
+            b::StaticMatrix{<:Any, <:Any, Tb}) where {Sb, Ta, Tb}
+            d = det(a)
+            T = typeof((one(Ta) * zero(Tb) + one(Ta) * zero(Tb)) / d)
+            if isbitstype(T)
+                # This if block can be removed when https://github.com/JuliaArrays/StaticArrays.jl/pull/749 is merged.
```
Comment (ChrisRackauckas) [inline src/linalg/linsolve.jl:52]: why is this related?

Comment (utkarsh530) [reply src/linalg/linsolve.jl:52]: I copied it from their implementation, I think it requires upstream changes to get it working.
